src/code/audio/audio_synthesis.c

```diff
@@ -1,137 +1,1705 @@
+            i++;
+        }
+
+        if (useReverb) {
+            if ((reverb->filterLeft != NULL) || (reverb->filterRight != NULL)) {
+                cmd = AudioSynth_FilterReverb(cmd, numSamplesPerUpdate * SAMPLE_SIZE, reverb);
+            }
+
+            // Saves the wet channel sample from DMEM (DMEM_WET_LEFT_CH) into (ringBuffer) DRAM for future use
+            if (subDelay != 0) {
+                cmd = AudioSynth_SaveSubReverbSamples(cmd, reverb, updateIndex);
+            } else {
+                if (reverb->mixReverbIndex != -1) {
+                    cmd = AudioSynth_MixOtherReverbIndex(cmd, reverb, updateIndex);
+                }
+                cmd = AudioSynth_SaveReverbSamples(cmd, reverb, updateIndex);
+            }
+        }
+    }
+
+    while (i < noteCount) {
+        cmd = AudioSynth_ProcessSample(
+            noteIndices[i], &gAudioContext.sampleStateList[sampleStateOffset + noteIndices[i]],
+            &gAudioContext.notes[noteIndices[i]].synthesisState, aiBuf, numSamplesPerUpdate, cmd, updateIndex);
+        i++;
+    }
+
+    size = numSamplesPerUpdate * SAMPLE_SIZE;
+    aInterleave(cmd++, DMEM_TEMP, DMEM_LEFT_CH, DMEM_RIGHT_CH, size);
+
+    if (gCustomAudioSynthFunction != NULL) {
+        cmd = gCustomAudioSynthFunction(cmd, 2 * size, updateIndex);
+    }
+    aSaveBuffer(cmd++, DMEM_TEMP, aiBuf, 2 * size);
+
+    return cmd;
+}
+
+Acmd* AudioSynth_ProcessSample(s32 noteIndex, NoteSampleState* sampleState, NoteSynthesisState* synthState, s16* aiBuf,
+                               s32 numSamplesPerUpdate, Acmd* cmd, s32 updateIndex) {
+    s32 pad1[2];
+    void* reverbAddrSrc;
+    Sample* sample;
+    AdpcmLoop* loopInfo;
+    s32 numSamplesUntilEnd;
+    s32 numSamplesInThisIteration;
+    s32 sampleFinished;
+    s32 loopToPoint;
+    s32 flags;
+    u16 frequencyFixedPoint;
+    s32 gain;
+    s32 frameIndex;
+    s32 skipBytes;
+    void* combFilterState;
+    s32 numSamplesToDecode;
+    s32 numFirstFrameSamplesToIgnore;
+    u8* sampleAddr;
+    u32 numSamplesToLoadFixedPoint;
+    s32 numSamplesToLoadAdj;
+    s32 numSamplesProcessed;
+    s32 sampleEndPos;
+    s32 numSamplesToProcess;
+    s32 dmemUncompressedAddrOffset2;
+    s32 pad2[3];
+    s32 numSamplesInFirstFrame;
+    s32 numTrailingSamplesToIgnore;
+    s32 pad3[3];
+    s32 frameSize;
+    s32 numFramesToDecode;
+    s32 skipInitialSamples;
+    s32 zeroOffset;
+    u8* samplesToLoadAddr;
+    s32 numParts;
+    s32 curPart;
+    s32 sampleDataChunkAlignPad;
+    s32 haasEffectDelaySide;
+    s32 numSamplesToLoadFirstPart;
+    u16 sampleDmemBeforeResampling;
+    s32 sampleAddrOffset;
+    s32 combFilterDmem;
+    s32 dmemUncompressedAddrOffset1;
+    Note* note;
+    u32 numSamplesToLoad;
+    u16 combFilterSize;
+    u16 combFilterGain;
+    s16* filter;
+    s32 bookOffset;
+    s32 finished;
+    s32 sampleDataChunkSize;
+    s16 sampleDataDmemAddr;
```

Comment:
Wow that's a scary stack.

Reply:
Looks like at least `bookOffset` and `finished` could be defined where declared.